src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Constant.cs

```diff
@@ -44,5 +45,68 @@ public long Int64Value
                 }
             }
         }
+
+        public override object Object
+        {
+            get
+            {
+                if (Type is NullType)
+                {
+                    return null;
+                }
+
+                object objval;
+                switch (System.Type.GetTypeCode(Type.AssociatedSystemType))
```

Comment:
@OmarTawfik moving this switch to `ExprConstant` is what I suggested would be ideal in reply to your comment at https://github.com/dotnet/corefx/pull/26414#discussion_r163115728 but for requiring dependencies on `TypeManager` that in other types' equivalent that would go away if `TypeManager` was static.